pkg/config/remote/service/service.go

```diff
@@ -276,10 +276,33 @@ func (s *Service) ClientGetConfigs(request *pbgo.ClientGetConfigsRequest) (*pbgo
 	if err != nil {
 		return nil, err
 	}
+
+	directorTargets, err := s.uptane.Targets()
+	if err != nil {
+		return nil, err
+	}
+	configPointers, err := executeClientPredicates(request.Client, directorTargets)
+	if err != nil {
+		return nil, err
+	}
+
+	// filter files to only return the configPointers ones
+	matchedConfigsMap := make(map[string]interface{})
+	for _, configPointer := range configPointers {
+		matchedConfigsMap[configPointer] = struct{}{}
+	}
```

Comment:
*flips table because Go makes you have to do this silly song and dance every time you want a set from a slice* 😆

Reply:
I heard there are solutions for this, starts with T I think